### PR TITLE
Re-add Rephrase Modal and add loading to question card buttons

### LINKED ISSUE #138: [BUG] Question card buttons can be spammed
**Describe the bug**

Fix: Add loading states to queue card buttons. Should help improve usability and prevent spam tapping

### PR DESCRIPTION
**Pull request by AdamFipke**
# Description

image
image

Closes #138 and #139

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This requires a run of `yarn install`

# How Has This Been Tested?

Manually tested each button

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed 
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any work that this PR is dependent on has been merged into the main branch
- [ ] Any UI changes have been checked to work on desktop, tablet, and mobile

### CODE REVIEW

File: packages/frontend/app/(dashboard)/course/[cid]/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  const { cid } = params
+
+  return (
+    <>
+      <AlertsContainer courseId={Number(cid)} />
+      {children}
+    </>
+  )
+}
+
```
Comment (AdamFipke) [inline packages/frontend/app/(dashboard)/course/[cid]/layout.tsx:23]: this basically means that alerts will be visible from any course page

File: packages/frontend/app/(dashboard)/course/[cid]/queue/[qid]/components/TAQuestionCardButtons.tsx [hunk 2]
```diff
@@ -347,8 +402,11 @@ const TAQuestionCardButtons: React.FC<TAQuestionCardButtonsProps> = ({
                   // clearTimeout(timerCheckout.current);
                   helpStudent()
                 }}
-                disabled={!canHelp}
+                disabled={
+                  !canHelp || rephraseButtonLoading || deleteButtonLoading
+                }
                 className="flex items-center justify-center"
+                loading={helpButtonLoading}
```
Comment (AdamFipke) [inline packages/frontend/app/(dashboard)/course/[cid]/queue/[qid]/components/TAQuestionCardButtons.tsx:409]: disable other buttons while one is loading

File: packages/frontend/app/(dashboard)/course/[cid]/queue/[qid]/components/modals/StudentRephraseModal.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      ]}
+      closable={false}
+    >
+      You have been requested to add more detail to your question by a member of
+      the course staff. While you elaborate on your question your place in line
+      will be reserved.
+    </Modal>
+  )
+}
+export default StudentRephraseModal
```
Comment (AdamFipke) [inline packages/frontend/app/(dashboard)/course/[cid]/queue/[qid]/components/modals/StudentRephraseModal.tsx:33]: This is basically all code from the old frontend

File: packages/frontend/app/components/AlertsContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          />
+        )
+    }
+  })
+
+  // probably want some better way of handling multiple alerts
+  return <div>{alertDivs}</div>
+}
+
+export default AlertsContainer
```
Comment (AdamFipke) [inline packages/frontend/app/components/AlertsContainer.tsx:46]: same with this

File: packages/frontend/app/components/AlertsContainer.tsx
Comment (wskksw) [inline packages/frontend/app/components/AlertsContainer.tsx:1]: ah nice, it uses the existing alerts framework

**Review by wskksw — APPROVED** [review]
good job

**Pull request merged**